### PR TITLE
enhance(docs): reorg google auth guide (web)

### PR DESCRIPTION
**Pull request by charislam**
- rearrange the page
- revalidate and revise the web instructions

### CODE REVIEW

File: apps/docs/content/guides/auth/social-login/auth-google.mdx [hunk 1]
```diff
@@ -516,66 +171,473 @@ Before you can use Sign in with Google, you need to obtain a [Google Cloud Platf
...
+   ```
+
+   Note that the nonce should be the same in both places, but because Supabase Auth expects the provider to hash it (SHA-256, hexadecimal representation), you need to provide a hashed version to Google and a non-hashed version to `signInWithIdToken`.
+
+   You can get both versions by using the in-built `crypto` library:
+
+   ```js
+   // Adapted from https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest#converting_a_digest_to_a_hex_string
+
+   const nonce = btoa(
```
Comment (charislam) [inline apps/docs/content/guides/auth/social-login/auth-google.mdx:303]: I tweaked this to work browser-side, would appreciate a second set of eyes

File: apps/docs/content/guides/auth/social-login/auth-google.mdx [hunk 2]
```diff
@@ -516,66 +171,473 @@ Before you can use Sign in with Google, you need to obtain a [Google Cloud Platf
...
+       nonce: '<NONCE>',
+     })
+   }
+   ```
+
+   Note that the nonce should be the same in both places, but because Supabase Auth expects the provider to hash it (SHA-256, hexadecimal representation), you need to provide a hashed version to Google and a non-hashed version to `signInWithIdToken`.
+
+   You can get both versions by using the in-built `crypto` library:
+
+   ```js
```
Comment (charislam) [inline apps/docs/content/guides/auth/social-login/auth-google.mdx:300]: Rewrote this to work browser-side since that seems to fit more with the example (script tags, functions in browser global scope). Would appreciate a second set of 👀 !

File: apps/docs/content/guides/auth/social-login/auth-google.mdx [hunk 3]
```diff
@@ -1,26 +1,19 @@
 ---
 id: 'auth-google'
 title: 'Login with Google'
-description: 'Use Sign in with Google on the web, in native apps or with Chrome extensions'
+description: 'Use Sign In with Google on the web, in native apps or with Chrome extensions'
```
Comment (hf) [inline apps/docs/content/guides/auth/social-login/auth-google.mdx:4]: Is this the new styling we apply? Maybe we should leave it for another PR where all `Sign in` is renamed to `Sign In`?

Your call!

File: apps/docs/content/guides/auth/social-login/auth-google.mdx (hunk 2 above)
Comment (hf) [reply apps/docs/content/guides/auth/social-login/auth-google.mdx:300]: I'm not sure I would class it as "browser side" since it's actually the olden days server side. But sure we can try it. I don't think I've seen anyone in Supabase use it.

**Review by hf — APPROVED** [review]
LGTM.

File: apps/docs/content/guides/auth/social-login/auth-google.mdx (hunk 3 above)
Comment (charislam) [reply apps/docs/content/guides/auth/social-login/auth-google.mdx:4]: Yeah I'll revert this 👍🏼

Comment (charislam) [inline apps/docs/content/guides/auth/social-login/auth-google.mdx:4]:
```suggestion
description: 'Use Sign in with Google on the web, in native apps or with Chrome extensions'
```
